packages/ui-react/src/containers/AccountModal/forms/PersonalDetails.tsx

```diff
@@ -44,13 +65,25 @@ const PersonalDetails = () => {
     navigate(modalURLFromLocation(location, hasOffers ? 'choose-offer' : 'welcome'), { replace: true });
   }, [navigate, location, accessModel, hasMediaOffers]);
 
+  const closeModal = useCallback(() => {
+    navigate(modalURLFromLocation(location, null), { replace: true });
+  }, [location, navigate]);
+
   useEffect(() => {
-    if (data && (!data.isCaptureEnabled || !data.shouldCaptureBeDisplayed)) nextStep();
+    if (!data) return;
+
+    if (isPreviouslyRegistered) {
+      closeModal();
```

Comment:
This might be tricky. `closeModal` depends on the `location` which changes when the modal closes. This re-reruns the effect again. If you need the `closeModal` in an effect, you might want to use `useEventCallback` to make the reference stable.

```tsx
const closeModal = useEventCallback(() => {
  navigate(modalURLFromLocation(location, null), { replace: true });
})l
```

Reply:
Although `nextStep` is also using this pattern...

Reply:
Once the modal gets closed there's no effect to be re-run, the modal is completely unmounted.

Reply:
I'm resolving this discussion as obsolete, as I removed `closeModal` in my last commit.